include/bout/index_derivs_interface.hxx

```diff
@@ -0,0 +1,178 @@
+
+  // Check the input data is valid
+  {
+    TRACE("Checking inputs");
+    checkData(f);
+    checkData(vel);
+  }
+
+  // Define properties of this approach
+  const CELL_LOC allowedStaggerLoc = localmesh->getAllowedStaggerLoc(direction);
+
+  // Handle the staggering
+  const CELL_LOC inloc = f.getLocation(); // Input locations
+  const CELL_LOC vloc = vel.getLocation();
+  if (outloc == CELL_DEFAULT) {
+    outloc = inloc;
+  }
+  const STAGGER stagger = localmesh->getStagger(vloc, inloc, outloc, allowedStaggerLoc);
+
+  // Check for early exit
+  const int nPoint = localmesh->getNpoints(direction);
+
+  if (nPoint == 1) {
+    auto tmp = T(0., localmesh);
+    tmp.setLocation(outloc);
+    return tmp;
+  }
+
+  // Lookup the method
+  auto& derivativeStore = DerivativeStore<T>::getInstance();
+  typename DerivativeStore<T>::upwindFunc derivativeMethod;
+  derivativeMethod =
+      derivativeStore.getFlowDerivative(method, direction, stagger, derivType);
+
+  // Create the result field
+  T result(localmesh);
+  result.allocate(); // Make sure data allocated
+  result.setLocation(outloc);
+
+  // Apply method
+  derivativeMethod(vel, f, result, region);
+
+  // Check the result is valid
+  {
+    TRACE("Checking result");
+    checkData(result);
+  }
+
+  return result;
+}
+
+/// The main kernel used for all standard derivatives
+template <typename T, DIRECTION direction, DERIV derivType>
+T indexStandardDerivative(const T& f, CELL_LOC outloc, const std::string& method,
+                          REGION region) {
+  AUTO_TRACE();
+
+  // Checks
+  static_assert(std::is_base_of<Field2D, T>::value || std::is_base_of<Field3D, T>::value,
+                "indexStandardDerivative only works on Field2D or Field3D input");
+
+  static_assert(derivType == DERIV::Standard || derivType == DERIV::StandardSecond
+                    || derivType == DERIV::StandardFourth,
+                "indexStandardDerivative only works for derivType in {Standard, "
+                "StandardSecond, StandardFourth}");
+
+  auto* localmesh = f.getMesh();
+
+  // Check that the input variable has data
+  ASSERT1(f.isAllocated());
+
+  // Check the input data is valid
+  {
+    TRACE("Checking input");
+    checkData(f);
+  }
+
+  // Define properties of this approach
+  const CELL_LOC allowedStaggerLoc = localmesh->getAllowedStaggerLoc(direction);
+
+  // Handle the staggering
+  const CELL_LOC inloc = f.getLocation(); // Input location
+  if (outloc == CELL_DEFAULT) {
+    outloc = inloc;
+  }
+  const STAGGER stagger = localmesh->getStagger(inloc, outloc, allowedStaggerLoc);
+
+  // Check for early exit
+  const int nPoint = localmesh->getNpoints(direction);
+
+  if (nPoint == 1) {
+    auto tmp = T(0., localmesh);
+    tmp.setLocation(outloc);
+    return tmp;
+  }
+
+  // Lookup the method
+  auto& derivativeStore = DerivativeStore<T>::getInstance();
+  typename DerivativeStore<T>::standardFunc derivativeMethod;
+
+  derivativeMethod =
```

Comment:
Could this just be `auto derivativeMethod = `, skipping the declaration above?

Reply:
Yes it could now that we've removed the branches!